test/UnitWOPISaveAsWithEncodedFileName.cpp

```diff
@@ -0,0 +1,94 @@
+/* -*- Mode: C++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4; fill-column: 100 -*- */
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <config.h>
+
+#include <WopiTestServer.hpp>
+#include <Log.hpp>
+#include <Unit.hpp>
+#include <UnitHTTP.hpp>
+#include <helpers.hpp>
+#include <Poco/Net/HTTPRequest.h>
+#include <Poco/Util/LayeredConfiguration.h>
+
+class UnitWOPISaveAsWithEncodedFileName : public WopiTestServer
+{
+    enum class Phase
+    {
+        LoadAndSaveAs,
+        Polling
+    } _phase;
+
+public:
+    UnitWOPISaveAsWithEncodedFileName()
+        : WopiTestServer("UnitWOPISaveAsWithEncodedFileName")
+        , _phase(Phase::LoadAndSaveAs)
+    {
+    }
+
+    void assertPutRelativeFileRequest(const Poco::Net::HTTPRequest& request) override
+    {
+        // spec says UTF-7...
+        LOK_ASSERT_EQUAL(std::string("/path/to/hello+ACU-20world.pdf"), request.get("X-WOPI-SuggestedTarget"));
```

Comment:
Is "hello+ACU-20world.pdf" correct? Is that related to UTF-7 encoding, as the comment implies?

Reply:
yes it is correct and it is related to UTF-7

Reply:
we send utf-7 encoded to integration %20 becomes +ACU-20 and later it is decoded to %20